### ISSUE TITLE
[Bug] nuScenes extractor ignores official train/val scene lists and creates overlapping splits

### ISSUE THREAD

**Issue by YuemingMrdream** [KTH-RPL/OpenSceneFlow#44]
The current nuScenes extractor  dataprocess/extract_nus.py   appears to ignore the actual scene names in
`nuscenes.utils.splits.train` and `splits.val`.

Although the official split lists are passed to `process_logs()`, the worker
arguments are constructed only from the length of each list:

```python
args = sorted([
    (data_mode, data_dir, scene_num_id, output_dir)
    for scene_num_id in range(len(scene_list))
])
Then process_log() selects a scene by its position in the nuScenes metadata:
scene = nusc.scene[scene_num_id]
Therefore, scene_list is only used to determine how many scenes are
processed. Its actual scene names are discarded.   
This results in duplication between the validation set and the training set.

**Comment by Kin-Zhang** [comment]
Thanks for reporting. I think you are correct. 
After checking the code, the truckscenes hold the same nus data format and also need to be fixed.

I will pass the real scene names rather than num_id for this.
